src/Components/SmartComponents/Systems/SystemsTableToolbar.js

```diff
@@ -0,0 +1,182 @@
+import { withRouter } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+import { dataShape } from '../../../Helpers/MiscHelper';
+import { middlewareListener } from '../../../Utilities/ReducerRegistry';
+import selectAllCheckbox from '../../../Helpers/selectAllCheckboxHelper';
+import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components';
+import searchFilter from '../../PresentationalComponents/Filters/PrimaryToolbarFilters/SearchFilter';
+import { fetchSystems, optOutSystemsAction } from '../../../Store/Actions/Actions';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications';
+import { downloadFile } from '@redhat-cloud-services/frontend-components-utilities/files/helpers';
+import { kebabAction, exportConfig, buildActiveFilters, removeFilters } from '../../../Helpers/TableToolbarHelper';
+
+/*eslint camelcase: 0*/
+const SystemsTableToolbar = (props) => {
+    const [filter, setFilter] = useState(undefined);
+    const [opt_out, setOptOut] = useState(false);
+    const [page_size] = useState(10);
+    const [page, setPage] = useState(1);
+    const dispatch = useDispatch();
+
+    const onSelect = (payload) => {
+        const isInArray = props.selectedHosts.some(item => item === payload.data.id);
+
+        if (!isInArray) {
+            const newSelected = payload.data.selected ? [payload.data.id, ...props.selectedHosts]
+                : props.selectedHosts.filter(item => item !== payload.data.id);
+            props.handleSelect(false, newSelected);
+        }
+    };
+
+    // eslint-disable-next-line no-unused-vars
+    const [selectListener] = useState(middlewareListener.addNew({
+        on: 'SELECT_ENTITY',
+        callback: payload => onSelect(payload)
+    }));
+
+    useEffect(()=>{
+        props.setSelectedHosts([]);
+        dispatch(fetchSystems({ filter, page, page_size, opt_out }));
+    }, [opt_out]);
+
+    const createUrlParams = allParams => {
+        const params = { ...allParams };
+        const allowedParams = ['filter', 'page', 'page_size'];
+        Object.keys(allParams).forEach(
+            key =>
+                (params[key] === undefined || params[key] === '' || !allowedParams.includes(key) || params[key] === false) &&
+                delete params[key]
+        );
+        const queryString = qs.stringify(params);
+
+        setFilter(allParams.filter);
+        props.history.push('?' + queryString);
+    };
+
+    const downloadReport = format => {
+        const { payload } = fetchSystems({
+            opt_out,
+            filter,
+            data_format: format,
+            page: 1,
+            page_size: Number.MAX_SAFE_INTEGER
+        });
+        payload &&
+            payload.then(({ data: response }) => {
+                const data = format === 'json' ? JSON.stringify(response) : response;
+                return downloadFile(data, `systems-${new Date().toISOString()}`, format);
```

Comment:
make sure the filename are according to the patterns which are used here: 
https://github.com/RedHatInsights/vulnerability-ui/pull/193